pkg/agent/route/route_linux.go

```diff
@@ -2004,27 +2115,39 @@ func (c *Client) DeleteEgressRoutes(tableID uint32) error {
 	return nil
 }
 
-func (c *Client) AddEgressRule(tableID uint32, mark uint32) error {
+func (c *Client) AddEgressRule(tableID uint32, mark uint32, isIPv6 bool) error {
+	family := netlink.FAMILY_V4
+	if isIPv6 {
+		family = netlink.FAMILY_V6
+	}
 	rule := netlink.NewRule()
 	rule.Table = int(tableID)
 	rule.Mark = mark
 	rule.Mask = ptr.To(types.SNATIPMarkMask)
+	rule.Family = family
 	if err := c.netlink.RuleAdd(rule); err != nil {
 		return fmt.Errorf("error adding ip rule %v: %w", rule, err)
 	}
+	c.egressRules.Store(generateRuleKey(rule), rule)
 	return nil
 }
 
-func (c *Client) DeleteEgressRule(tableID uint32, mark uint32) error {
+func (c *Client) DeleteEgressRule(tableID uint32, mark uint32, isIPv6 bool) error {
```

Comment:
Can this handle the old rules properly when there is no family specified in old version?

Reply:
Yes, I have tested the code manually and found this.